src/utils/index.js

```diff
@@ -54,3 +54,67 @@ export function convertFormDataToObject( formData, includedFields = [] ) {
 		return acc;
 	}, {} );
 }
+
+/**
+ * Register a reader activity dispatch on an element event.
+ *
+ * @param {string|Element} element The element to register the activity on. Can either be the element node or a string for the element selector.
+ * @param {string}         action  The action to dispatch an activity for.
+ * @param {Function}       cb      The callback to populate the activity data.
+ * @param {string}         event   The element event to listen for. Defaults to `submit` for form elements and `click` for other elements.
+ */
+export function registerElementActivity( element, action, cb, event ) {
+	window.newspackRAS = window.newspackRAS || [];
+	if ( typeof element === 'string' ) {
+		element = [ ...document.querySelectorAll( element ) ];
+	}
+	if ( ! element ) {
+		return;
+	}
+	// If no callback is provided, use a noop.
+	if ( ! cb ) {
+		cb = () => ( {} );
+	}
+	event = event || ( element.tagName === 'FORM' ? 'submit' : 'click' );
+
+	if ( element && ! Array.isArray( element ) ) {
+		element = [ element ];
+	}
```

Comment:
```suggestion
	if ( element && ! Array.isArray( element ) ) {
		element = [ element ];
	}
	if ( ! element?.length ) {
		return;
	}
	// If no callback is provided, use a noop.
	if ( ! cb ) {
		cb = () => ( {} );
	}
	event = event || ( element[0].tagName === 'FORM' ? 'submit' : 'click' );
```

Reply:
Ah, good catch! I think it's safer to just place the event detection inside the loop. Updated in 39625d592733b4c35861f705833355cab02884bf

Reply:
I think I wasn't able to reproduce because I was just scrolling down and doing a single click on the checkout submit button 😅